### DISCUSSION TITLE
How to change route in utils file

### DISCUSSION THREAD

**Discussion by weisiqian** [nuxt/nuxt#18173] [Questions]
I have a util file. here is the code.

![image](https://user-images.githubusercontent.com/19994951/140601711-28460f45-0464-42ed-8a05-7f84023416ba.png)

there‘s anyway to change route in the function

**Comment by wobsoriano** [comment]
I think a composable would do?

```ts
const router = useRouter()
```

**Reply by weisiqian** [reply]
I tried to use useRouter() in composables directory and utils directory.

in composable directory

```
import { useRouter } from 'vue-router';

const router = useRouter()

export function navigateTo(url: string) {
  router.push(url)
}
```

in utils directory

```
import { useRouter } from 'vue-router';

const router = useRouter()

export function navigateTo(url: string) {
  router.push(url)
}
```

but it throws a undefined Error when I call navigateTo function in component 。

![image](https://user-images.githubusercontent.com/19994951/140602379-ab1613af-024b-4311-a2a5-d4c929a02d8b.png)

I create a mini repository here : [https://github.com/weisiqian/nuxt3-app.git](https://github.com/weisiqian/nuxt3-app.git)

**Reply by wobsoriano** [reply]
You're using `useRouter()` wrong

**Reply by weisiqian** [reply]
what should I do

**Reply by Fire7** [reply]
@weisiqian same for me. 
Can you please provide how to use router in plugins? 
I'm trying to get it inside `defineNuxtPlugin`, but it is undefined: 

```
  export default defineNuxtPlugin((nuxtApp) => {
    const router = useRouter() // router in undefined
```

**Reply by crotanite** [reply]
@Fire7 Inside a plugin you'll want to use the below:

```
const router = nuxtApp.$router
```
